### PR TITLE
[NRLF-396] Removed custodian:identifier from producer search endpoints

### PR DESCRIPTION
**Pull request by brianbayes**

### CODE REVIEW

File: api/producer/swagger.yaml [hunk 1]
```diff
@@ -1693,7 +1692,6 @@ components:
     RequestQueryCustodian:
       type: string
       pattern: ^https\:\/\/fhir\.nhs\.uk\/Id\/ods-organization-code\|(\w+)$
```
Comment (jaklinger) [inline api/producer/swagger.yaml:1694]: should this be removed altogether?

File: api/producer/nrl-producer-api.yaml [hunk 2]
```diff
@@ -1596,7 +1595,6 @@ components:
     RequestQueryCustodian:
       type: string
       pattern: ^https\:\/\/fhir\.nhs\.uk\/Id\/ods-organization-code\|(\w+)$
```
[the remark this answers is not in the document]
Comment (nomad3k) [reply api/producer/nrl-producer-api.yaml:1597]: We've got some more work in this area on the backlog to clarify the use of this field.  https://nhsd-jira.digital.nhs.uk/browse/NRLF-425

[the remark this answers is not in the document]
Comment (nomad3k) [reply api/producer/nrl-producer-api.yaml:1597]: I got confused about the use of `this` and thought you were referring to the regex and not the entire component definition.  Agreed, that the whole thing should be taken out root and branch.

[the remark this answers is not in the document]
Comment (nomad3k) [reply api/producer/nrl-producer-api.yaml:1597]: There's probably a `schema` and `parameter` definition I'd guess.

**Review by nomad3k — CHANGES_REQUESTED** [review]

**Review by nomad3k — APPROVED** [review]

**Pull request merged**